File: WEEK5/Symbol.java

import java.util.Map;
import java.util.Objects;
import java.util.Set;

//
public class Symbol extends Sentence {
    private String name;

    public Symbol(String name) {
        this.name = name;
    }

    @Override
    public boolean evaluate(Map<String, Boolean> model) {
        return model.get(name);
    }

    @Override
    public String formula() {
        return name;
    }

    @Override
    public Set<String> symbols() {
        return Set.of(name);
    }

    @Override
    public String toString() {
        return name;
    }

    @Override
    public boolean equals(Object o) {
        return (o instanceof Symbol s) &&
                s.name.equals(this.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name);
    }
}

File: WEEK6/Symbol.py
from Sentence import *


class Symbol(Sentence):
    """Represents a symbol in a logical sentence."""

    def __init__(self, name):
        """Initialize a symbol with a given name."""
        self.name = name

    def __eq__(self, other):
        """Check equality with another symbol."""
        return isinstance(other, Symbol) and self.name == other.name

    def __hash__(self):
        """Return hash of the symbol."""
        return hash(("symbol", self.name))

    def __repr__(self):
        """Return string representation of the symbol."""
        return self.name

    def evaluate(self, model):
        """Evaluate the symbol given a model."""
        return bool(model[self.name])

    def formula(self):
        """Return the formula representation of the symbol."""
        return self.name

    def symbols(self):
        """Return set containing this symbol."""
        return {self.name}
